### PR TITLE
fix numpy comparison

### PR DESCRIPTION
**Pull request by adriendelsalle**
Description
--

The numpy arrays comparison made in `DataUnion` causes :
- `DeprecationWarning` when the 2 arrays have different sizes/shapes
- observers to be triggered even when arrays are identical be cause the method used always raise an error

_from package traitlets/traitlets/traitlets.py#567_

``` python
silent = bool(old_value == new_value)
``` 

Related to this [pythreejs issue]( https://github.com/jupyter-widgets/pythreejs/issues/332).

Proposition
---
Overload the `set` method of `DataUnion`.

### CODE REVIEW

File: ipydatawidgets/ndarray/union.py [hunk 1]
```diff
@@ -36,6 +37,24 @@ def __init__(self, default_value=Undefined, dtype=None, shape_constraint=None,
...
+        except KeyError:
+            old_value = self.default_value
+
+        obj._trait_values[self.name] = new_value
+        try:
+            silent = np.array_equal(old_value, new_value)
+        except:
+            # if there is an error in comparing, default to notify
+            silent = False
+        if silent is not True:
```
Comment (vidartf) [inline ipydatawidgets/ndarray/union.py:53]: This should probably be `!= True`, as this also gives deprecation warnings for Python 3.8 and above

Comment (adriendelsalle) [reply ipydatawidgets/ndarray/union.py:53]: Hi @vidartf ! Thank you for your quick answer.
I'm not sure to understand your comment, is there something going wrong?
Could you please give me more details?

It works fine for me on Python3.8, running this simple example:

``` python
import warnings

from ipydatawidgets import DataUnion
from traitlets import HasTraits, observe
import numpy as np

warnings.simplefilter("always")

class TestClass(HasTraits):

    array = DataUnion()

    @observe('array')
    def print(self, changes):
        print(changes['new'])

test = TestClass()
test.array = np.array([10., 20., 30.], dtype=np.float32)
test.array = np.array([10., 40., 30.], dtype=np.float32)  # <-- was ok, still ok
test.array = np.array([10., 40., 30.], dtype=np.float32)  # <-- observer no more triggered when array is unchanged
test.array = np.array([10., 40.], dtype=np.float32)  # <-- no more DeprecationWarning
```

Comment (vidartf) [reply ipydatawidgets/ndarray/union.py:53]: Sorry, my mistake. Python 3.8 has a new warning for comparing literals with `is`, and I got confused and thought it included boolean literals as well. It does not :)

**Comment by adriendelsalle** [comment]
Thanks @vidartf!
Do you need anything more on this PR before considering to merge?

**Comment by vidartf** [comment]
No, thanks. I'm just waiting on myself to have the time to fix the Appveyor tests + do a release. I know from experience that on low-volume repos, if I just merge and do not the release immediately, it can easily be forgotten.

**Comment by vidartf** [comment]
I'm assuming a release will be needed for lab3 compatibility, but I'm waiting for a ipywidgets release to see what changes might be needed.

**Pull request merged**